### PR TITLE
CIF-1675: make Venia IT and UI tests pass on CM

### PR DESCRIPTION
**Pull request by Buuhuu**
## Description

Adjust the IT and UI tests in order to make them pass on CM.

## Related Issue

CIF-1675

## Motivation and Context

Automate various processes

## How Has This Been Tested?

CM production pipeline, CI, locally

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

File: it.tests/src/main/java/com/venia/it/tests/CommerceTestBase.java [hunk 1]
```diff
@@ -66,9 +46,21 @@
     public CQRule cqBaseRule = new CQRule(cqBaseClassRule.authorRule);
 
     protected static CQClient adminAuthor;
+    protected static boolean isCloudEnvironment;
 
     @BeforeClass
     public static void init() {
         adminAuthor = cqBaseClassRule.authorRule.getAdminClient(CommerceClient.class);
+        // check again if the environment may be a cloud environment (either skd on localhost, or an adobeaemcloud host)
+        isCloudEnvironment = adminAuthor.getUrl().getAuthority().contains("localhost")
```
Comment (herzog31) [inline it.tests/src/main/java/com/venia/it/tests/CommerceTestBase.java:55]: How does this condition work with a local or CircleCI AEM 6.5 instance?

On CircleCI for example, you can see that the 6.5 IT job skips the `testProductListPageWithSampleData65` and `testProductPageWithSampleData65` tests, which should not be the case.

Comment (Buuhuu) [reply it.tests/src/main/java/com/venia/it/tests/CommerceTestBase.java:55]: Right, I missed that. Then we cannot use this approach unfortunately.

Comment (Buuhuu) [reply it.tests/src/main/java/com/venia/it/tests/CommerceTestBase.java]: I removed those methods again and assert the canonical url with an `endsWith` instead.

**Comment by herzog31** [comment]
Since you removed the JUnit categories, we can also remove https://github.com/adobe/aem-cif-guides-venia/blob/main/.circleci/ci/it-tests.js#L57 and its usages.

**Review by herzog31 — APPROVED** [review]

**Pull request merged**